internal/api/graphql/access/token_auth.go

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package access
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type TokenAuth struct {
+	logger Logger
+	secret []byte
+}
+
+func NewTokenAuth(l Logger) *TokenAuth {
+	return &TokenAuth{logger: l, secret: []byte(os.Getenv("AUTH_TOKEN_SECRET"))}
+}
+
+type TokenClaims struct {
```

Comment:
What about [basic claims](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1) such as

* iss (Issuer): Identify the token issuer.
* sub (Subject): Identify the subject of the token (usually a user ID).
* exp (Expiration Time): Set an expiration time for the token.
* iat (Issued At): Include the time the token was issued.

?

Reply:
Done